vars/MavenCommand.groovy

```diff
@@ -0,0 +1,127 @@
+def class MavenCommand {
+
+    def steps
+
+    boolean returnStdout = false
+
+    String settingsXmlPath = ''
+    List mavenOptions = []
+    Map properties = [:]
+    String logFileName = null
+    List profiles = []
+
+    MavenCommand(steps){
+        this.steps = steps
+    }
+
+    MavenCommand(steps, List defaultOpts){
+        this(steps)
+        this.mavenOptions.addAll(defaultOpts)
+    }
+
+    def run(String goals) {
```

Comment:
What if we refactor this method by (just a suggestion)?
```
def run(String goals) {
        def cmdBuilder = new StringBuilder("mvn -B")
        if(this.settingsXmlPath) {
            cmdBuilder.append(" -s ${this.settingsXmlPath}")
        }
        if(this.mavenOptions.size() > 0){
            cmdBuilder.append(' ').append(this.mavenOptions.join(' '))
        }
        cmd.append(' ').append(goals)
        if(this.profiles.size() > 0){
            cmdBuilder.append(' -P').append(this.profiles.join(','))
        }
        if(this.properties.size()){
            cmdBuilder.append(' ').append(this.properties.collect{ it.value != '' ? "-D${it.key}=${it.value}" : "-D${it.key}" }.join(' '))
        }
        if(this.logFileName){
            cmdBuilder.append(" | tee \$WORKSPACE/${logFileName} ; test \${PIPESTATUS[0]} -eq 0")
        }

        return steps.sh(script: cmdBuilder.toString(), returnStdout: this.returnStdout)
    }
```

Reply:
Using a MavenBuilder is indeed better. gonna change that ;)